### PR TITLE
Test Windows CI

### PR DESCRIPTION
**Pull request by mamantoha**
This was just a test for Windows CI

```
Finished in 49.14 seconds
121 examples, 13 failures, 16 errors, 0 pending
```

### CODE REVIEW

**Comment by mamantoha** [comment]
to fix the ameba workflow go to Setting -> Actions -> General and check "Read and write permissions"

image

**Comment by sdogruyol** [comment]
@mamantoha that's already enabled, do I need to do anything else?

**Comment by mamantoha** [comment]
@sdogruyol I don't understand why `check_ameba` fails. It works fine on my fork.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -33,9 +36,20 @@ jobs:
         run: |
           crystal spec
           crystal spec --release --no-debug
+        continue-on-error: ${{ matrix.allowed-failure }}
 
       - name: Check formatting
         run: crystal tool format --check
```
Comment (straight-shoota) [inline .github/workflows/ci.yml:42]: This fails on Windows, not sure why. It works fine locally.
I would suggest to disable this spec on windows. Actually, it only needs to execute once - the expected result is the same on any runner (`if: matrix.os == 'ubuntu-latest' && matrix.crystal == 'nightly'`)

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -33,9 +36,20 @@ jobs:
...
+    steps:
+      - name: Install Crystal
+        uses: crystal-lang/install-crystal@v1
+      - name: Check out repository code
+        uses: actions/checkout@v3
+      - name: Crystal Ameba Linter
+        id: crystal-ameba
+        uses: crystal-ameba/github-action@v0.7.1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (straight-shoota) [inline .github/workflows/ci.yml:55]: The action seems to be broken (ref https://github.com/kemalcr/kemal/actions/runs/5443532334/jobs/9900164974?pr=657). So better exclude that from this PR. It's not necesary for Windows support.

**Review by straight-shoota — COMMENTED** [review]
It would be great to get Windows support merged. 
On the code side the only relevant change is `Process.on_interrupt` - and that is direly needed. Maybe this could be merged on its own if CI needs more time?

**Comment by HertzDevil** [comment]
The `Kemal::StaticFileHandler` specs are failing because of this part:

https://github.com/kemalcr/kemal/blob/c995a2a9711d72d90b347683007b45bbaf495317/src/kemal/static_file_handler.cr#L30-L38

Both `@public_dir` and `expanded_path` contain the drive name, so `file_path` becomes something like `C:\...\C:\...`. Replacing the first line with just `expanded_path = request_path` makes all specs pass. I have not dug far enough to understand if the `File.expand_path` call has any significance on other platforms.

**Comment by straight-shoota** [comment]
`StaticFileHandler` probably just needs to be aligned with stdlib's implementation.

**Comment by HertzDevil** [comment]
That part in stdlib is:

```crystal
    request_path = Path.posix(request_path)
    expanded_path = request_path.expand("/")

    file_path = @public_dir.join(expanded_path.to_kind(Path::Kind.native))
```

**Comment by mamantoha** [comment]
Hi @HertzDevil @straight-shoota I don't have a Windows environment to test this build locally. Can someone else take over and continue working on this pull request? Thank you.

File: shard.yml [hunk 3]
```diff
@@ -11,11 +11,7 @@ dependencies:
   exception_page:
     github: crystal-loot/exception_page
     version: ~> 0.3.0
-
-development_dependencies:
-  ameba:
-    github: crystal-ameba/ameba
-    version: ~> 1.4.0
+    branch: master
```
Comment (Sija) [inline shard.yml:14]: This should be reverted, I believe.

**Comment by mamantoha** [comment]
Closed in favour of https://github.com/kemalcr/kemal/pull/690

**Pull request closed without merging**